src/DefaultUnleash.php

```diff
@@ -49,6 +52,18 @@ public function isEnabled(string $featureName, ?Context $context = null, bool $d
             return $default;
         }
 
+        if (method_exists($feature, 'hasImpressionData') && $feature->hasImpressionData()) {
+            $event = new ImpressionDataEvent(
+                ImpressionDataEventType::IS_ENABLED,
+                Uuid::v4(),
+                clone $this->configuration,
+                clone $context,
+                clone $feature,
```

Comment:
It makes sense to me to send a copy and not the actual object so that it cannot be changed from within the event. In php `clone` performs a shallow copy. From [php documentation](https://www.php.net/manual/en/language.oop5.cloning.php):

> When an object is cloned, PHP will perform a shallow copy of all of the object's properties. Any properties that are references to other variables will remain references.

Reply:
Definitely agree, a shallow copy is probably okay here